docker/index.js

```diff
@@ -16,86 +20,197 @@ const ain = new AinJs(blockchainEndpoint, chainId);
-        parsedInputPath[6] !== 'input') {
-        console.log(`Not supported path pattern: ${inputPath}`);
-        return;
-    }
-
-    // 2. call GET /tasks/{task_id}
-    const inputValue = tx.tx_body.operation.value;
-    const options = JSON.parse(inputValue);
-
-    const task_id = options.task_id;
-    const pickedOptions = (({ prompt, seed, guidance_scale }) => ({ prompt, seed, guidance_scale }))(options);
-    pickedOptions = {...pickedOptions, ...{"num_images_per_prompt": 1}};
-    
-    // pickedOptions = {
-    //     prompt: ...,
-    //     seed: ...,
-    //     guidance_scale: ...,
-    //     num_images_per_prompt: 1
-    // }
-    
-    const SDResult = await axios.get(`${SD_INPAINTING_ENDPOINT}/tasks/${task_id}`, pickedOptions);
-    console.log(JSON.stringify(SDResult.data,null,2));
-
-    if (SDResult.data.status !== "completed") {
-        console.log(`Task ${task_id} is not completed!`);
-        res.send(`Task ${task_id} is not completed!`);
-        return;
-    }
-    
-
-    //pre-check the output path
-    const outputPath = formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "signed_data"]);
-    const result = await ain.db.ref(outputPath).setValue({
-      value: `${JSON.stringify(SDResult.data, null, 2)}`,
-      nonce: -1,
-    })
-    .catch((e) => {
-      console.error(`setValue failure:`, e);
-    });
-  
-});
+	const { transaction } = req.body;
+	const value = transaction.tx_body.operation.value;
+	const taskId = value.task_id;
+	let uploadMetadataRes; // for catch error
+
+	const inputPath = transaction.tx_body.operation.ref;
+	const parsedInputPath = parsePath(inputPath);
+
+	// pre-check the output path
+	const outputPath = await formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "signed_data"]);
+
+	// init pinata sdk
+	const pinata = new pinataSDK({ pinataApiKey, pinataSecretApiKey });
+
+
+	// get generated ainft image url with with task id
+	const result = await axios.get(`${process.env.SD_INPAINTING_ENDPOINT}/tasks/${taskId}`);
+
+	const imageUrls = result.data.result;
+
+	// get image file from url
+	const imageDataResponse = await axios.get(imageUrls[value.params.index || 1], {
+		responseType: "arraybuffer",
+	});
+
+	// for test
+	const imageBuffer = fs.readFileSync(__dirname + "/test.png");
+
+	// image file to readable stream
+	const imageDataStream = new Readable();
+	imageDataStream.push(imageBuffer);
+	imageDataStream.push(null);
+
+	// upload image to pinata
+	const options = {
+		pinataMetadata: {
+			name: `${taskId}_image`,
+		},
+		pinataOptions: {
+			cidVersion: 0,
+		},
+	};
+
+	// upload image to ipfs
+	const uploadImgRes = await pinata.pinFileToIPFS(imageDataStream, options).catch(err => console.error(err));
+
+	// get origin metadata
+	const originMetadata = await axios.get(`https://gateway.pinata.cloud/ipfs/${value.contract_info.metadata_cid}/${value.contract_info.token_id}`, {
+		headers: {
+			'Accept': '*/*'
+		}
+	});
```

Comment:
I don't quite understand what originMetadata is, is the metadata of the initially issued NFT accessible that way?
If so, will this code be continuously available as we continue to update the metadata?
(It might be more comfortable doing this in Video Call.)